docs/docsite/rst/presentations/openstack-modules-intro.rst

```diff
@@ -0,0 +1,394 @@
+
+  * os_security_group
+  * os_floating_ip
+
+OpenStack Dynamic Inventory Script
+==================================
+
+* github.com/ansible/ansible/contrib/inventory/openstack.py
+* Uses all hosts in all clouds as one inventory
+* Excludes hosts without IPs
+* Makes a bunch of auto-groups (flavor, image, az, region, cloud)
+* Makes groups for `metadata['groups']`
+
+Modules for All OpenStack Resources are Welcome Upstream
+========================================================
+
+* If there is a thing you want to manage without a module, add one
+* We're very welcoming/accepting
+* Same goes for features to existing modules
+
+To Serve All Users, We Do Have to be Strict
+===========================================
+
+* Modules that touch OpenStack APIs must do so via `shade`
+* Other modules don't need to (ex: `os_tempest_config`)
+* In user modules, vendor differences should be hidden
+* In operator modules, exposing them is ok
+
+clouds.yaml
+===========
+
+Information about the clouds you want to connect to is stored in a file
+called `clouds.yaml`.
+
+Both `yaml` and `yml` are acceptable.
+
+`clouds.yaml` can be in your homedir: `~/.config/openstack/clouds.yaml`
+or system-wide: `/etc/openstack/clouds.yaml`.
+
+Information in your homedir, if it exists, takes precedence.
+
+Full docs on `clouds.yaml` are at
+https://docs.openstack.org/developer/os-client-config/
+
+What about Mac and Windows?
+===========================
+
+`USER_CONFIG_DIR` is different on Linux, OSX and Windows.
+
+* Linux: `~/.config/openstack`
+* OSX: `~/Library/Application Support/openstack`
+* Windows: `C:\\Users\\USERNAME\\AppData\\Local\\OpenStack\\openstack`
+
+`SITE_CONFIG_DIR` is different on Linux, OSX and Windows.
+
+* Linux: `/etc/openstack`
+* OSX: `/Library/Application Support/openstack`
+* Windows: `C:\\ProgramData\\OpenStack\\openstack`
+
+Only for Ansible
+================
+
+/etc/ansible/clouds.yaml
+
+Config Terminology
+==================
+
+For multi-cloud, think of two types:
+
+* `profile` - Facts about the `cloud` that are true for everyone
+* `cloud` - Information specific to a given `user`
+
+Remember your Execution Context!
+================================
+
+* ansible executes code on remote systems
+* clouds.yaml needs to be on the host where the modules run
+* auth information can be passed to modules directly
+* other config (currently) MUST go in `clouds.yaml`
+* **TODO**: Expose ability to pass entire cloud config
+
+basic clouds.yaml for the example code
+======================================
+
+Simple example of a clouds.yaml
+
+* Config for a named `cloud` "my-citycloud"
+* Reference a well-known "named" profile: `citycloud`
+* `os-client-config` has a built-in list of profiles at
+  https://docs.openstack.org/developer/os-client-config/vendor-support.html
+* Vendor profiles contain various advanced config
+* `cloud` name can match `profile` name (using different names for clarity)
```

Comment:
If these need to show up with codename fonts, the syntax is ``cloud`` and ``profile``